Userland/Libraries/LibGUI/FileSystemModel.cpp

```diff
@@ -629,7 +629,9 @@ Icon FileSystemModel::icon_for(Node const& node) const
     return FileIconProvider::icon_for_path(node.full_path(), node.mode);
 }
 
+// Null pointers indicate an image that couldn't be loaded due to errors.
 static HashMap<DeprecatedString, RefPtr<Gfx::Bitmap>> s_thumbnail_cache;
+static HashMap<DeprecatedString, NonnullRefPtr<Threading::BackgroundAction<ErrorOr<NonnullRefPtr<Gfx::Bitmap>>>>> s_loading_thumbnails;
```

Comment:
Nesting template parameters that deep suggests that perhaps a new type might be a good idea.

As a starting point I suggest `using BitmapBackgroundAction = Threading::BackgroundAction<ErrorOr<NonnullRefPtr<Gfx::Bitmap>>>;`